### PR TITLE
Add multi-DC support for Kubernetes integration

### PR DESCRIPTION
**Pull request by adelbertc**
Also add support for Kubernetes versions 1.6.x and 1.7.x.

### CODE REVIEW

File: core/src/main/scala/Config.scala [hunk 1]
```diff
@@ -540,12 +540,16 @@ object Config {
         })
     }
 
-    def readKubernetesInfrastructure(kfg: KConfig): Option[Infrastructure.Kubernetes] = {
-      (kfg.lookup[String]("endpoint") |@| kfg.lookup[Duration]("timeout")) { (endpoint, timeout) =>
+    def readKubernetesInfrastructure(kfg: KConfig): Option[Infrastructure.Kubernetes] =
+      (kfg.lookup[String]("endpoint") |@|
+       kfg.lookup[String]("version").flatMap(KubernetesVersion.fromString) |@|
+       kfg.lookup[String]("caCert").map(p => Paths.get(p)) |@|
```
Comment (timperrett) [inline core/src/main/scala/Config.scala]: we've avoided camel case for configuration paramaters, in favor of kebab case. Can we rename this to `ca-cert`?

File: core/src/main/scala/Config.scala [hunk 2]
```diff
@@ -559,12 +563,8 @@ object Config {
         def checkServerTrusted(certs: Array[X509Certificate], authType: String): Unit = ()
       }
 
-    def getKubernetesPodCert(): Option[SSLContext] = {
-      // Auto-mounted at this path for pods
-      // https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/#accessing-the-api-from-a-pod
-      val certBundle = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
-
-      val is = new FileInputStream(certBundle)
+    def getKubernetesPodCert(certPath: Path): Option[SSLContext] = {
```
Comment (timperrett) [inline core/src/main/scala/Config.scala]: Just so I understand what's going on here, you're assuming that there's per-cluster PKI for interacting with the Kube API? Presumably this means the location that Nelson is deployed needs the certs on disk (this should be fine, just trying to grok it)

Comment (adelbertc) [reply core/src/main/scala/Config.scala]: Yep that is the current assumption

File: core/src/main/scala/Config.scala (hunk 1 above)
Comment (adelbertc) [reply core/src/main/scala/Config.scala]: kebab-case, love it. will do

**Pull request merged**